octodns/manager.py

```diff
@@ -244,6 +227,37 @@ def __init__(self, config_file, max_workers=None, include_meta=False):
                     'Incorrect plan_output config for ' + plan_output_name
                 )
 
+        self._zone_tree = None
+
+    @property
+    def zone_tree(self):
+        if self._zone_tree is None:
+            zone_tree = {}
+
+            # Get a list of all of our zone names
+            zones = list(self.config['zones'].keys())
+            # Sort them from shortest to longest so that parents will always
+            # come before their subzones
+            zones.sort(key=lambda z: len(z))
+            # Until we're done processing zones
+            while zones:
+                # Grab the one we'lre going to work on now
+                zone = zones.pop(0)
```

Comment:
Removing an element from the front of the list is `O(n)` expensive isn't it?
Maybe collections.deque would be a good fit here?

Reply:
On paper it's a problem, but given that we're really only working with sizes on the order of 0-10 it's negligible. Since deque is part of the std lib it seems reasonable to switch to it though.